packages/babel-traverse/src/path/removal.js

```diff
@@ -2,13 +2,14 @@
 
 import { hooks } from "./lib/removal-hooks";
 import { REMOVED, SHOULD_SKIP } from "./index";
+import * as t from "@babel/types";
 
-export function remove() {
+export function remove(node?: Object) {
```

Comment:
Is it possible to fix this without modifying the public API? Maybe we could remove the bindings in `replaceWithMultiple` before setting `this.node` to `null`.

Alternatively, we could leave `@babel/traverse` as-is (this PR doesn't fix the problem anyway, for example when removing `if (foo) var bar;`), and check if the binding exists when using it.

Reply:
> check if the binding exists when using it

Does it mean it's expected that `this._removeFromScope()` does not remove all bindings, and all plugins should be aware of that?

Then I'll add a check for `node.removed` in `plugin-transform-paramters`.

Reply:
Yes, `_removeFromScope` is only expected to remove bindings of the current declaration (`this.node`).

Maybe we could fix it by calling `this._removeFromScope()` in `path.replaceWithMultiple`, before setting `this._node` to `null`?

> all plugins should be aware of that?

Unfortunately yes :man_shrugging: :slightly_frowning_face: 
Until we manage to fully automate our scope handling, plugins should be aware that it could not be in sync with the AST.